packages/block-library/src/image/index.php

```diff
@@ -0,0 +1,357 @@
+		// Note: there is a constrain on the largest image size in the srcset attribute, see `wp_calculate_image_srcset()`.
+		$full_file_width = (int) $image_meta['width'];
+	} else {
+		$full_file_width = $image_file_width;
+	}
+
+	if ( $block_attributes['align'] === 'full' ) {
+		$image_width = min( (int) $block_witdh['full'], $full_file_width );
+	} elseif ( $block_attributes['align'] === 'wide' ) {
+		$image_width = min( (int) $block_witdh['wide'], $full_file_width );
+	} else {
+		$image_width = (int) $block_witdh['default'];
+
+		if ( ! empty( $block_attributes['width'] ) && ! empty( $block_attributes['editWidth'] ) ) {
+			// Scale the image if it was resized in the editor.
+			$scale = $block_attributes['width'] / $block_attributes['editWidth'];
+			// TODO: allow upscaling? Looks ugly, but someone may expect it?
+			$image_width = round( $image_width * $scale );
+		}
+
+		$image_width = min( $image_width, $image_file_width );
+	}
+
+	// Recalculate.
+	$image_size = wp_constrain_dimensions( $image_file_width, $image_file_height, $image_width );
+
+	/**
+	 * Filters the calculated image size for the image block.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @param array $image_size The calculated image size width and height (in that order).
+	 * @param array $block_attributes The block attributes.
+	 * @param array $image_meta The image attachment meta data.
+	 */
+	return apply_filters( 'gutenberg_get_image_width_height', $image_size, $block_attributes, $image_meta );
+}
+
+function render_block_core_image( $block_attributes = array(), $html = '' ) {
+	// Old post or... something's wrong.
+	if ( empty( $html ) || empty( $block_attributes ) ) {
+		return $html;
+	}
+
+	$image_src = '';
+	$attachment_id = 0;
+	$image_meta = null;
+	$srcset = '';
+	$sizes = '';
+
+	// Local image
+	if ( ! empty( $block_attributes['id'] ) ) {
+		$attachment_id = (int) $block_attributes['id'];
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+		// Broken or very old image meta? Incorrect attachment id?
+		if ( empty( $image_meta['height'] ) || empty( $image_meta['width'] ) ) {
+			return $html;
+		}
+
+		if ( empty( $block_attributes['url'] ) ) {
+			// Old post. Proceed if the full size image was used (default old behavior).
+			$attachment_url = wp_get_attachment_url( $attachment_id );
+
+			if ( strpos( $html, ' src="' . $attachment_url . '"' ) !== false ) {
+				// If the full size image was used, attempt to use the large size, if it has the same ratio.
+				if ( ! empty( $image_meta['sizes']['large'] ) ) {
+					$large = $image_meta['sizes']['large'];
+
+					if ( wp_image_matches_ratio( $large['width'], $large['height'], $image_meta['width'], $image_meta['height'] ) ) {
+						$image_data = wp_get_attachment_image_src( $attachment_id, 'large', false );
+
+						// Can be `false`.
+						if ( $image_data ) {
+							$block_attributes['url'] = $image_data[0];
+							// wp_get_attachment_image_src() calls image_downsize()
```

Comment:
Have we changed coding standards for multi-line comments in PHP?

Reply:
I've always referred to this: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-2-multi-line-comments

Reply:
Yeahhh, I'm (slowly) reviewing some of the coding standards. Seems there have been a lot of additions and restrictions that do not fully comply with the "actual" WordPress coding standards :)

This is one of these rules. Using `*` makes it look too close to a DocBlock which is wrong.